### PR TITLE
SCUS-63 Add EVENT_INDEXER support

### PR DESCRIPTION
**Pull request by MohamedFarouk-HMCTS**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -33,10 +33,10 @@
     <properties>
         <cpp.repo.name>framework-generators</cpp.repo.name>
 
-        <common-bom.version>1.28.0</common-bom.version>
-        <framework-api.version>4.0.0-M9</framework-api.version>
-        <framework.version>6.0.0-M15</framework.version>
-        <generator-maven-plugin.version>2.7.0-M1</generator-maven-plugin.version>
+        <common-bom.version>1.29.0</common-bom.version>
+        <framework-api.version>4.0.0-SNAPSHOT</framework-api.version>
+        <framework.version>6.0.0-M10</framework.version>
```
Comment (MartinYSpasov) [inline pom.xml]: You have to release framework first

File: pom.xml [hunk 2]
```diff
@@ -33,10 +33,10 @@
     <properties>
         <cpp.repo.name>framework-generators</cpp.repo.name>
 
-        <common-bom.version>1.28.0</common-bom.version>
-        <framework-api.version>4.0.0-M9</framework-api.version>
-        <framework.version>6.0.0-M15</framework.version>
-        <generator-maven-plugin.version>2.7.0-M1</generator-maven-plugin.version>
+        <common-bom.version>1.29.0</common-bom.version>
+        <framework-api.version>4.0.0-SNAPSHOT</framework-api.version>
```
Comment (MartinYSpasov) [inline pom.xml]: add latest framework-api version

File: generators/messaging-adapter-generator/src/main/java/uk/gov/justice/raml/jms/core/ComponentDestinationType.java [hunk 3]
```diff
@@ -40,6 +42,11 @@ public ComponentDestinationType() {
      *                                  Component}
      */
     public Class<? extends Destination> inputTypeFor(final String component) {
+
+        if (components.containsKey(component)) {
```
Comment (arcadius) [inline generators/messaging-adapter-generator/src/main/java/uk/gov/justice/raml/jms/core/ComponentDestinationType.java]: Had to move this up as IMHO, this is the correct location.
When it was down, it never get hit during tests as both `COMMAND_API` and `CUSTOM_COMMAND_API` will get fulfilled by the condition 
```
  if (component.contains(COMMAND_API)) {
            return components.get(COMMAND_API);
        }
```

File: pom.xml [hunk 4]
```diff
@@ -33,9 +33,9 @@
     <properties>
         <cpp.repo.name>framework-generators</cpp.repo.name>
 
-        <common-bom.version>1.28.0</common-bom.version>
-        <framework-api.version>4.0.0-M9</framework-api.version>
-        <framework.version>6.0.0-M15</framework.version>
+        <common-bom.version>1.29.0</common-bom.version>
```
Comment (allanmckenzie) [inline pom.xml]: common-bom version should be 1.28.0

**Pull request closed without merging**